libcaf_core/test/detail/rfc3629.cpp

```diff
@@ -122,43 +123,87 @@ constexpr std::string_view ascii_2 = R"__(
  *                      \____/_/   \_|_|                                      *
 )__";
 
-} // namespace
+TEST("ASCII input") {
+  check(valid_utf8(ascii_1));
+  check(valid_utf8(ascii_2));
+}
+
+TEST("valid UTF-8 input") {
+  check(valid_utf8(valid_two_byte_1));
+  check(valid_utf8(valid_two_byte_2));
+  check(valid_utf8(valid_three_byte_1));
+  check(valid_utf8(valid_three_byte_2));
+  check(valid_utf8(valid_four_byte_1));
+  check(valid_utf8(valid_four_byte_2));
+}
+
+TEST("invalid UTF-8 input") {
+  check(!valid_utf8(invalid_two_byte_1));
+  check(!valid_utf8(invalid_two_byte_2));
+  check(!valid_utf8(invalid_two_byte_3));
+  check(!valid_utf8(invalid_two_byte_4));
+  check(!valid_utf8(invalid_three_byte_1));
+  check(!valid_utf8(invalid_three_byte_2));
+  check(!valid_utf8(invalid_three_byte_3));
+  check(!valid_utf8(invalid_three_byte_4));
+  check(!valid_utf8(invalid_three_byte_5));
+  check(!valid_utf8(invalid_three_byte_6));
+  check(!valid_utf8(invalid_three_byte_7));
+  check(!valid_utf8(invalid_three_byte_8));
+  check(!valid_utf8(invalid_four_byte_1));
+  check(!valid_utf8(invalid_four_byte_2));
+  check(!valid_utf8(invalid_four_byte_3));
+  check(!valid_utf8(invalid_four_byte_4));
+  check(!valid_utf8(invalid_four_byte_5));
+  check(!valid_utf8(invalid_four_byte_6));
+  check(!valid_utf8(invalid_four_byte_7));
+  check(!valid_utf8(invalid_four_byte_8));
+  check(!valid_utf8(invalid_four_byte_9));
+  check(!valid_utf8(invalid_four_byte_10));
+}
+
+TEST("invalid UTF-8 input missing continuation bytes") {
+  check_eq(rfc3629::validate(invalid_two_byte_1), res_t(0, true));
+  check_eq(rfc3629::validate(invalid_three_byte_1), res_t{0ul, true});
```

Comment:
`res_t` has the type information, so we no longer need the `ul` suffies.

```suggestion
  check_eq(rfc3629::validate(invalid_three_byte_1), res_t{0, true});
```